contrib/rrdtool/patches/0007-fix-libxml-cfi.patch

```diff
@@ -0,0 +1,369 @@
+diff --git a/src/rrd_restore.c b/src/rrd_restore.c
+index 14b9446b..9f94f2df 100644
+--- a/src/rrd_restore.c
++++ b/src/rrd_restore.c
+@@ -48,6 +48,11 @@ static int opt_force_overwrite = 0;
+  * Helpers
+  */
+ 
++__attribute__((no_sanitize("cfi-icall")))
```

Comment:
I was told that @q66 probably wouldn't approve this :) But right now the alternatives are either patching libxml2 (to export memory management functions as real functions instead of `extern` function pointers) or disabling CFI altogether.

Reply:
this is technically fine, but i'd probably just disable CFI altogether here, it uses glib (as well as libxml2) and CFI with glib things is largely a futile effort (you can be about 80% sure any glib callback is function pointer UB) and overly fragile